pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -0,0 +1,779 @@
+            pulsar.getLocalMetadataStore().registerSessionListener(this::handleMetadataSessionEvent);
+            log.debug("Successfully registered the handleMetadataSessionEvent");
+
+            log.info("Successfully started the channel.");
+        } catch (Exception e) {
+            String msg = "Failed to start the channel.";
+            log.error(msg, e);
+            throw new PulsarServerException(msg, e);
+        }
+    }
+
+
+    public void close() throws PulsarServerException {
+        try {
+            if (leaderElectionService != null) {
+                leaderElectionService.close();
+                leaderElectionService = null;
+                log.debug("Successfully closed the channel leader election service.");
+            }
+
+            if (tableview != null) {
+                tableview.close();
+                tableview = null;
+                log.debug("Successfully closed the channel tableview.");
+            }
+
+            if (producer != null) {
+                producer.close();
+                producer = null;
+                log.info("Successfully closed the channel producer.");
+            }
+
+            // TODO: clean brokerRegistry
+
+            if (cleanupTasks != null) {
+                cleanupTasks.cancel(true);
+                cleanupTasks = null;
+                log.info("Successfully cancelled the cleanup tasks");
+            }
+
+            log.info("Successfully closed the channel.");
+
+        } catch (Exception e) {
+            String msg = "Failed to close the channel.";
+            log.error(msg, e);
+            throw new PulsarServerException(msg, e);
+        }
+    }
+
+    public void scheduleCompaction() throws PulsarServerException {
+        try {
+            Long threshold = pulsar.getAdminClient().topicPolicies()
+                    .getCompactionThreshold(TOPIC);
+            if (threshold == null || threshold == 0) {
+                pulsar.getAdminClient().topicPolicies()
+                        .setCompactionThreshold(TOPIC, COMPACTION_THRESHOLD);
+                log.info("Scheduled compaction on topic:{}, threshold:{} bytes", TOPIC, COMPACTION_THRESHOLD);
+            } else {
+                log.info("Already set compaction on topic:{}, threshold:{} bytes", TOPIC, COMPACTION_THRESHOLD);
+            }
+        } catch (PulsarAdminException e) {
+            throw new PulsarServerException("Failed to schedule compaction.", e);
+        }
+    }
+
+    public String getChannelOwner() {
+        CompletableFuture<Optional<LeaderBroker>> future = leaderElectionService.readCurrentLeader();
+        if (!future.isDone()) {
+            return null;
+        }
+        Optional<LeaderBroker> leader = future.join();
+        if (leader.isEmpty()) {
+            return null;
+        }
+        //expecting http://broker-xyz:port
+        String broker = leader.get().getServiceUrl();
+        broker = broker.substring(broker.lastIndexOf('/') + 1);
```

Comment:
we should have some utility method to do this, maybe it is out of the scope of this PS

Reply:
Agreed, but I assume this util func is not required in this PR. I added a TODO comment here. 

`// TODO: discard this protocol prefix removal by a util func that returns lookupServiceAddress(serviceUrl)`

Reply:
+1